deepchem/metrics/__init__.py

```diff
@@ -181,7 +182,12 @@ def compute_metric(self, y_true, y_pred, w=None, n_classes=2, filter_nans=True):
       if filter_nans:
         computed_metrics = np.array(computed_metrics)
         computed_metrics = computed_metrics[~np.isnan(computed_metrics)]
-      return self.task_averager(computed_metrics)
+      if self.compute_force_metrics:
```

Comment:
Done. I think this should be moved into user space by making it a `task_averager` function. Not to be done in this PR, but I've added a line note as a comment.